core/trino-main/src/test/java/io/trino/sql/query/TestCheckConstraint.java

```diff
@@ -0,0 +1,193 @@
+
+@TestInstance(PER_CLASS)
+@Execution(SAME_THREAD)
+public class TestCheckConstraint
+{
+    private static final String LOCAL_CATALOG = "local";
+    private static final String MOCK_CATALOG = "mock";
+    private static final String USER = "user";
+
+    private static final Session SESSION = testSessionBuilder()
+            .setCatalog(LOCAL_CATALOG)
+            .setSchema(TINY_SCHEMA_NAME)
+            .setIdentity(Identity.forUser(USER).build())
+            .build();
+
+    private QueryAssertions assertions;
+
+    @BeforeAll
+    public void init()
+    {
+        LocalQueryRunner runner = LocalQueryRunner.builder(SESSION).build();
+
+        runner.createCatalog(LOCAL_CATALOG, new TpchConnectorFactory(1), ImmutableMap.of());
+
+        MockConnectorFactory mock = MockConnectorFactory.builder()
+                .withGetColumns(schemaTableName -> {
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation"))) {
+                        return TPCH_NATION_SCHEMA;
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_multiple_column_constraint"))) {
+                        return TPCH_NATION_SCHEMA;
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_invalid_function"))) {
+                        return TPCH_NATION_SCHEMA;
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_not_boolean_expression"))) {
+                        return TPCH_NATION_SCHEMA;
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_subquery"))) {
+                        return TPCH_NATION_SCHEMA;
+                    }
+                    throw new UnsupportedOperationException();
+                })
+                .withCheckConstraints(schemaTableName -> {
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation"))) {
+                        return ImmutableList.of("nationkey > 100");
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_multiple_column_constraint"))) {
+                        return ImmutableList.of("nationkey > 100 AND regionkey > 50");
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_invalid_function"))) {
+                        return ImmutableList.of("invalid_function(nationkey) > 100");
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_not_boolean_expression"))) {
+                        return ImmutableList.of("1 + 1");
+                    }
+                    if (schemaTableName.equals(new SchemaTableName("tiny", "nation_subquery"))) {
+                        return ImmutableList.of("nationkey > (SELECT count(*) FROM nation)");
```

Comment:
Naive question: Would there a comparison with the output of  `INSERT` be also a "valid" expression?

```
nationkey > (INSERT INTO nation VALUES (....))
``

Reply:
It's not a valid expression. The internal logic calls `io.trino.sql.parser.SqlParser#createExpression` that expects boolean expression.